gobblin-cluster/src/main/java/gobblin/cluster/GobblinClusterManager.java

```diff
@@ -149,15 +167,109 @@ public GobblinClusterManager(String clusterName, String applicationId, Config co
   }
 
   /**
+   * Handle leadership change.
+   * The applicationLauncher is only started on the leader.
+   * The leader cleans up existing jobs before starting the applicationLauncher.
+   * @param changeContext notification context
+   */
+  private void handleLeadershipChange(NotificationContext changeContext) {
+    if (this.helixManager.isLeader()) {
```

Comment:
Should we ensure that the current leader has stopped before starting services here?